File: bms/go/geo/geo_s1_Interval_InteriorIntersects/geo_s1_Interval_InteriorIntersects.go
package test

import (
	"math"
)

func (i geo_s1_Interval) InteriorIntersects(oi geo_s1_Interval) bool {
	if i.IsEmpty() || oi.IsEmpty() || i.Lo == i.Hi {
		return false
	}
	if i.IsInverted() {
		return oi.IsInverted() || oi.Lo < i.Hi || oi.Hi > i.Lo
	}
	if oi.IsInverted() {
		return oi.Lo < i.Hi || oi.Hi > i.Lo
	}
	return (oi.Lo < i.Hi && oi.Hi > i.Lo) || i.IsFull()
}

func (i geo_s1_Interval) IsEmpty() bool { return i.Lo == math.Pi && i.Hi == -math.Pi }

func (i geo_s1_Interval) IsInverted() bool { return i.Lo > i.Hi }

func (i geo_s1_Interval) IsFull() bool { return i.Lo == -math.Pi && i.Hi == math.Pi }

type geo_s1_Interval struct {
	Lo, Hi float64
}

File: translations_only/success/go/geo/geo_s1_Interval_InteriorIntersects/mistral/Restart/geo_s1_Interval_InteriorIntersects.rs

use std::f64;

struct GeoS1Interval {
    lo: f64,
    hi: f64,
}

impl GeoS1Interval {
    fn is_empty(&self) -> bool {
        self.lo == f64::consts::PI && self.hi == -f64::consts::PI
    }

    fn is_inverted(&self) -> bool {
        self.lo > self.hi
    }

    fn is_full(&self) -> bool {
        self.lo == -f64::consts::PI && self.hi == f64::consts::PI
    }

    fn interior_intersects(&self, oi: &GeoS1Interval) -> bool {
        if self.is_empty() || oi.is_empty() || self.lo == self.hi {
            return false;
        }
        if self.is_inverted() {
            return oi.is_inverted() || oi.lo < self.hi || oi.hi > self.lo;
        }
        if oi.is_inverted() {
            return oi.lo < self.hi || oi.hi > self.lo;
        }
        (oi.lo < self.hi && oi.hi > self.lo) || self.is_full()
    }
}
